src/malli/core.cljc

```diff
@@ -350,10 +350,12 @@
         (if (== ci i)
           {:children (arr->vec -children)
            :entries (arr->vec -entries)
-           :forms (arr->vec -forms)}
+           :forms (arr->vec -forms)
+           :keyset (#?(:clj set, :cljs deref) -keyset)}
```

Comment:
Should we coerce to a set at this point?
It's used only for membership check internally. Can we just skip it?

Reply:
Then instead of `contains?` for `clj` we can call `.contains`

Reply:
tested quickly: keeping the HashSet is few times faster than using a PersistentHashSet. Coercing into a PersistentHashSet was the slowest. I think the persistent hashmap is a good start as the result is visible to the outside world, eg. part of the extender api (`-parse-entries`).

Reply:
@ikitommi this might be a bit of an overkill but we can wrap the mutable hash set with a light weight immutable wrapper which behaves externally like a set:
```clojure
(defn- -set-wrapper
  [^java.util.Set s]
  (reify
    clojure.lang.IPersistentSet
    (disjoin [_ key] (disj (set s)))
    (contains [_ key] (.contains key))
    (get [_ key] (.get s key))
    clojure.lang.Counted
    (count [_] (.size s))
    clojure.lang.IPersistentCollection
    (cons [_ o] (cons o (seq s)))
    (empty [_] #{})
    (equiv [_ o] (.equiv (set s) o))
    clojure.lang.Seqable
    (seq [_] (seq s))
    clojure.lang.IFn
    (invoke [_ o] (when (.contains s o) o))))
```